CHANGELOG.md

```diff
@@ -1,10 +1,43 @@
+* The [push service is deprecated](https://www.mongodb.com/docs/atlas/app-services/reference/push-notifications/) from the servers point of view, we've deprecated this on v11 and removed it from v12.
+* We’ve decided to remove numeric indexing and “array methods” from the SubscriptionSet, since it was would bloat our SDK code, the team see little actual use-case for it, we expect very few developers actually using this and the workaround is really easy (spreading into an array `[...subscriptions]`). Again something we would love feedback on.
+* No longer exporting the `ObjectPropsType`, `UserMap`, `UserType`, `BaseFunctionsFactory`, `AuthProviders`, `PropertyType`, `HTTP`, `*Details` interfaces of the `EmailPasswordAuthClient` and `AuthError` types, since it wasn’t used internally, not expected to be used by users and most are very simple to type out for any user relying on it. Similarly the `DictionaryBase` is a type that was introduced to help work around an issue (declaring string index accessors on a class with methods) in our declarations. We consider it an internal detail that got introduced as part of our public API by accident. We propose removing this and ask users to simply use the `Dictionary` type directly. We also decided to rename the `Session` class to `SyncSession` since it’s now exported directly on the package namespace. `Session` will still be available (but deprecated) on the `Realm.Sync.Session`. We’re no longer using the `*Payload` types (were actually only used by Realm Web) and we don’t expect end-users to be relying directly on these, hence they will be deleted.
+* The return values of Object#getPropertyType was changed to return `"list"` instead of `"array"`.
 
 ### Enhancements
-* Added configuration option `SyncConfiguration.cancelWaitsOnNonFatalError`. Set to true, all async operations (such as opening the Realm with `Realm.open`) will fail when a non-fatal error, such as a timeout, occurs.  
+* The new SDK wraps its accessor objects in a Proxy trapping the ownKeys trap which enables calls to the standard `Object.keys(obj)` and the spread operator `{...obj}`
+* Added configuration option `SyncConfiguration.cancelWaitsOnNonFatalError`. Set to true, all async operations (such as opening the Realm with `Realm.open`) will fail when a non-fatal error, such as a timeout, occurs.
 * Added an overload to `Object.linkingObjects` method that takes type of the linking object as an input instead of its string name ([#5326](https://github.com/realm/realm-js/issues/5326))  
```

Comment:
Can we go through the whole document and pick a consistent way to refer to `Realm.Object` (or `RealmObject`, or `realm.Object`, ...)? I don't think we should use unqualified `Object` since that clashes with the js `Object` constructor object. We probably should consider renaming it in our own code as well for the same reason.

Reply:
It's called `RealmObject` in our own code, except before the final export for backwards compatibility (and because it's shorter).